plugins/hls-cabal-plugin/src/Ide/Plugin/Cabal/Completion/Completer/Snippet.hs

```diff
@@ -0,0 +1,77 @@
+{-# LANGUAGE OverloadedStrings #-}
+
+module Ide.Plugin.Cabal.Completion.Completer.Snippet where
+
+import           Control.Monad                                (forM)
+import           Data.Function                                ((&))
+import           Data.Map                                     (Map)
+import qualified Data.Map                                     as Map
+import           Data.Maybe                                   (fromMaybe)
+import qualified Data.Text                                    as T
+import           Distribution.Compat.Lens                     ((?~))
+import           Ide.Plugin.Cabal.Completion.Completer.Types
+import           Ide.Plugin.Cabal.Completion.Types
+import qualified Language.LSP.Protocol.Lens                   as JL
+import qualified Language.LSP.Protocol.Types                  as LSP
+
+import           Ide.Plugin.Cabal.Completion.Completer.Simple
+import qualified Text.Fuzzy.Parallel                          as Fuzzy
+
+-- | Maps snippet triggerwords with their completers
+snippetCompleter :: Completer
+snippetCompleter _ cData = do
+  let scored = Fuzzy.simpleFilter 1000 10 (completionPrefix prefInfo) $ Map.keys snippetMap
+  forM
+    scored
+    (\compl -> do
+        let matched = Fuzzy.original compl
+        let completion = fromMaybe [] $ Map.lookup matched snippetMap
+        pure $ mkSnippetCompletion (T.unlines completion) matched
+    )
+      where
+        prefInfo = cabalPrefixInfo cData
+        mkSnippetCompletion :: T.Text -> T.Text -> LSP.CompletionItem
+        mkSnippetCompletion insertText toDisplay = mkDefaultCompletionItem toDisplay
+            & JL.kind ?~ LSP.CompletionItemKind_Snippet
+            & JL.insertText ?~ insertText
+            & JL.insertTextFormat ?~ LSP.InsertTextFormat_Snippet
+        snippetMap :: Map T.Text [T.Text]
+        snippetMap = Map.fromList
```

Comment:
Should be its own function. Avoids this right drift.

How about a type def for trigger words? `Map TriggerWord T.Text`